### PR TITLE
cmake, doc: Update `build-unix.md`

### PR DESCRIPTION
**Pull request by hebasto**

### CODE REVIEW

File: doc/build-unix.md [hunk 1]
```diff
@@ -195,19 +192,19 @@ Additional Configure Flags
...
-    ./configure --help
+    cmake -B build -LH
 
 
 Setup and Build Example: Arch Linux
 -----------------------------------
 This example lists the steps necessary to setup and build a command line only distribution of the latest changes on Arch Linux:
 
-    pacman --sync --needed autoconf automake boost gcc git libevent libtool make pkgconf python sqlite
+    pacman --sync --needed cmake boost gcc git libevent libtool make pkgconf python sqlite
```
Comment (pablomartin4btc) [inline doc/build-unix.md:202]: nit: do we still need `libtool` there?
```suggestion
    pacman --sync --needed cmake boost gcc git libevent make pkgconf python sqlite
```

**Review by pablomartin4btc — APPROVED** [review]
utACK 52400af574c36d447c0aaa492565b85f51dde66d

Checked consistency also against #201.

File: doc/build-unix.md (hunk 1 above)
Comment (hebasto) [reply doc/build-unix.md]: Thanks! Fixed.

**Comment by pablomartin4btc** [comment]
re-ACK a11cb27e2969377dfdae591463d86f7bc91a742f

File: doc/build-unix.md [hunk 2]
```diff
@@ -8,10 +8,9 @@ To Build
 ---------------------
 
 ```bash
-./autogen.sh
-./configure
-make # use "-j N" for N parallel jobs
-make install # optional
+cmake -B build
```
Comment (m3dwards) [inline doc/build-unix.md:11]: Might it be nice to add a comment that these commands are run from the root of the project's source code?

File: doc/build-unix.md [hunk 3]
```diff
@@ -134,7 +133,7 @@ GUI dependencies:
 
 If you want to build bitcoin-qt, make sure that the required packages for Qt development
 are installed. Qt 5 is necessary to build the GUI.
-To build without GUI pass `--without-gui`.
+To build with GUI pass `-DBUILD_GUI=ON`.
 
 To build with Qt 5 you need the following:
```
Comment (m3dwards) [inline doc/build-unix.md:139]: "Once these are installed, they will be found by configure and a bitcoin-qt executable will be
built by default."

This is no longer true?

File: doc/build-unix.md [hunk 4]
```diff
@@ -44,7 +44,7 @@ Finally, clang (often less resource hungry) can be used instead of gcc, which is
 
 Build requirements:
 
-    sudo apt-get install build-essential libtool autotools-dev automake pkg-config bsdmainutils python3
+    sudo apt-get install build-essential cmake pkg-config bsdmainutils python3
 
 Now, you can either build from self-compiled [depends](#dependencies) or install the required dependencies:
```
Comment (m3dwards) [inline doc/build-unix.md:49]: Is it worth quickly mentioning for those who are not familiar with depends that depends is for static linking?

**Comment by m3dwards** [comment]
Added some nits, two unrelated to the change to cmake.

File: doc/build-unix.md (hunk 4 above)
Comment (hebasto) [reply doc/build-unix.md:49]: I'm not oppose to this suggestion, but this PR is focused on rather switching build systems than overall improving.

File: doc/build-unix.md (hunk 2 above)
Comment (hebasto) [reply doc/build-unix.md:11]: I'm not oppose to this suggestion, but this PR is focused on rather switching build systems than overall improving.

However, I'll support such an improvement once it is PR'ed to the main repository.

File: doc/build-unix.md (hunk 3 above)
Comment (m3dwards) [reply doc/build-unix.md:139]: So this being the case, shouldn't we delete the line: "Once these are installed, they will be found by configure and a bitcoin-qt executable will be built by default."

Comment (hebasto) [reply doc/build-unix.md:136]: Thanks! Deleted.

**Comment by hebasto** [comment]
@m3dwards's https://github.com/hebasto/bitcoin/pull/208#discussion_r1609874775 has been addressed.

File: doc/build-unix.md [hunk 5]
```diff
@@ -195,19 +189,19 @@ Additional Configure Flags
 --------------------------
 A list of additional configure flags can be displayed with:
 
-    ./configure --help
+    cmake -B build -LH
```
Comment (TheCharlatan) [inline doc/build-unix.md:192]: Just a suggestion: We could also mention `ccache` here.

File: doc/build-unix.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hebasto) [reply doc/build-unix.md:11]: Indeed. There was `-S` in the past. See:
- https://github.com/hebasto/bitcoin/pull/167#discussion_r1579455791
- https://github.com/hebasto/bitcoin/pull/190

We agreed to use modern CMake invocation style. Configuring in the source root makes the `-S` option redundant.

File: doc/build-unix.md (hunk 5 above)
Comment (hebasto) [reply doc/build-unix.md:192]: Mind elaborating your suggestion?

FWIW, there is _"Cache compilations with `ccache`"_ section in https://github.com/bitcoin/bitcoin/blob/master/doc/productivity.md.

Comment (TheCharlatan) [reply doc/build-unix.md:192]: Just an unrelated passing comment. I guess it makes more sense in productivity.md. Please resolve.

**Pull request merged**